File: vertx-wechat-sdk/src/main/java/tk/okou/vertx/sdk/tencent/wechat/WechatMiniProgramApi.java
package tk.okou.vertx.sdk.tencent.wechat;

import io.vertx.codegen.annotations.Fluent;
import io.vertx.codegen.annotations.VertxGen;
import io.vertx.core.AsyncResult;
import io.vertx.core.Handler;
import io.vertx.core.Vertx;
import io.vertx.core.json.JsonObject;
import tk.okou.vertx.sdk.tencent.BaseMiniProgramApi;
import tk.okou.vertx.sdk.tencent.wechat.impl.WechatMiniProgramApiImpl;
@VertxGen
public interface WechatMiniProgramApi extends WechatMiniApi, BaseMiniProgramApi {
    static WechatMiniProgramApi create(Vertx vertx, WechatMiniApiOptions options) {
        return new WechatMiniProgramApiImpl(vertx, options);
    }

    static WechatMiniProgramApi create(Vertx vertx) {
        return new WechatMiniProgramApiImpl(vertx, new WechatMiniApiOptions());
    }

    @Fluent
    WechatMiniProgramApi code2session(String appId, String secret, String jsCode, Handler<AsyncResult<JsonObject>> handler);

    @Fluent
    WechatMiniProgramApi code2session(String appId, String secret, String jsCode, String grantType, Handler<AsyncResult<JsonObject>> handler);

    @Fluent
    WechatMiniProgramApi getAccessToken(String appId, String secret, Handler<AsyncResult<JsonObject>> handler);

    @Fluent
    WechatMiniProgramApi getAccessToken(String grantType, String appId, String secret, Handler<AsyncResult<JsonObject>> handler);

}

File: vertx-wechat-sdk-kotlin-coroutines/src/main/kotlin/tk/okou/vertx/sdk/tencent/kotlin/wechat/WechatMiniProgramApi.kt
/*
 * Copyright 2019 Red Hat, Inc.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * and Apache License v2.0 which accompanies this distribution.
 *
 * The Eclipse Public License is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * The Apache License v2.0 is available at
 * http://www.opensource.org/licenses/apache2.0.php
 *
 * You may elect to redistribute this code under either of these licenses.
 */
package tk.okou.vertx.sdk.tencent.kotlin.wechat

import io.vertx.core.json.JsonObject
import io.vertx.kotlin.coroutines.awaitResult
import tk.okou.vertx.sdk.tencent.wechat.WechatMiniProgramApi

suspend fun WechatMiniProgramApi.code2accessTokenAwait(appId: String, secret: String, jsCode: String): JsonObject {
  return awaitResult {
    this.code2session(appId, secret, jsCode, it)
  }
}

suspend fun WechatMiniProgramApi.code2accessTokenAwait(appId: String, secret: String, jsCode: String, grantType: String): JsonObject {
  return awaitResult {
    this.code2session(appId, secret, jsCode, grantType, it)
  }
}

suspend fun WechatMiniProgramApi.getAccessTokenAwait(appId: String, secret: String): JsonObject {
  return awaitResult {
    this.getAccessToken(appId, secret, it)
  }
}

suspend fun WechatMiniProgramApi.getAccessTokenAwait(grantType: String, appId: String, secret: String): JsonObject {
  return awaitResult {
    this.getAccessToken(grantType, appId, secret, it)
  }
}

